client/ayon_maya/plugins/publish/extract_layout.py

```diff
@@ -112,10 +111,12 @@ def process(self, instance):
 
             t = transform.translation(om.MSpace.kWorld)
             t = om.MVector(t.x, t.z, -t.y)
+            s = transform.scale(om.MSpace.kObject)
+            s = [s[0], s[2], s[1]]
+            r = transform.rotation()
             transform.setTranslation(t, om.MSpace.kWorld)
-            transform.rotateBy(
-                om.MEulerRotation(math.radians(-90), 0, 0), om.MSpace.kWorld)
-            transform.scaleBy([1.0, 1.0, -1.0], om.MSpace.kObject)
+            transform.setRotation(om.MEulerRotation(r.x, r.z, r.y))
+            transform.setScale(s, om.MSpace.kObject)
```

Comment:
What is it that we're trying to do - are we trying to rotate an object in world-space around the pivot point?
Isn't that just a matrix multiplication?

Reply:
As maya by default is Y-axis up and unreal is Z-axis up, what we are trying to do is make sure transformation data which are Y-axis up converting to the data which are Z-axis up. Would be helpful if you can come up with some hints(or codes) for the matrix multiplication. We can edit the identity matrix from the basis, of which we dont need to setScale and setTranslation like this.  But for the rotation is more complicated issue, it really needs some maths to be done.